src/theme-provider.js

```diff
@@ -0,0 +1,75 @@
+import React, {PropTypes, Component} from 'react'
+import brcast from 'brcast'
+
+export const CHANNEL = '__glamorous__'
+
+/**
+ * This is a component which will provide a theme to the entire tree
+ * via context and event listener (because pure components block context updates)
+ * @param {Object} theme the theme object..
+ */
+class ThemeProvider extends Component {
+  broadcast = brcast(this.props.theme)
+
+  // create theme, by merging with outer theme, if present
+  getTheme = passedTheme => {
```

Comment:
This is always called on the instance `this.getTheme` so it doesn't need to be a public class field and can be a normal method.